app/controllers/api/physical_servers_controller.rb

```diff
@@ -52,26 +52,47 @@ def refresh_resource(type, id, _data = nil)
       end
     end
 
-    private
-
-    def change_resource_state(state, type, id)
-      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+    def apply_config_pattern_resource(type, id, data)
+      change_resource_state(:apply_config_pattern, type, id, data)
+    end
 
-      ensure_resource_exists(type, id) if single_resource?
+    private
 
+    def change_resource_state(state, type, id, data = {})
+      validate_resource(state, type, id, data)
       api_action(type, id) do |klass|
         begin
           server = resource_search(id, type, klass)
+          ensure_customization_script_exists(data["pattern_id"]) if apply_config_action(state)
           desc = "Requested server state #{state} for #{server_ident(server)}"
           api_log_info(desc)
           task_id = queue_object_action(server, desc, :method_name => state, :role => :ems_operations)
           action_result(true, desc, :task_id => task_id)
-        rescue => err
+        rescue StandardError => err
           action_result(false, err.to_s)
         end
       end
     end
 
+    def validate_resource(state, type, id, data = {})
+      raise BadRequestError, "Must specify an id for changing a #{type} resource" unless id
+      if single_resource?
+        ensure_resource_exists(type, id)
```

Comment:
I do not believe this call is necessary, because when you do a `resource_search`, it will raise the appropriate message if the resource does not exist

Reply:
Yes. I agree. I'll make the changes